internal/controller/compaction/reconciler.go

```diff
@@ -187,10 +206,106 @@ func isJobPresent(job *batchv1.Job) bool {
 	return job != nil && job.Name != ""
 }
 
+// waitForJobDeletionMarker waits for the job to have a deletion timestamp set or to be completely deleted.
+// This ensures that the next reconciliation will not see a stale job without deletion marker due to informer cache staleness.
+func (r *Reconciler) waitForJobDeletionMarker(ctx context.Context, logger logr.Logger, jobName, jobNamespace string) error {
+	logger.Info("Waiting for job deletion marker to be set", "jobName", jobName, "namespace", jobNamespace)
+
+	err := wait.PollUntilContextTimeout(ctx, pollInterval, pollTimeout, true,
+		func(ctx context.Context) (bool, error) {
+			job, err := r.fetchCompactionJob(ctx, jobName, jobNamespace)
+			if err != nil {
+				logger.Error(err, "Error fetching job while waiting for deletion marker",
+					"jobName", jobName, "namespace", jobNamespace)
+				return false, err // This will cause retry
+			}
+
+			if job == nil {
+				logger.Info("Job completely deleted", "jobName", jobName, "namespace", jobNamespace)
+				return true, nil // Job is gone, success
+			}
+
+			if !job.DeletionTimestamp.IsZero() {
+				logger.Info("Job deletion marker is set", "jobName", jobName, "namespace", jobNamespace)
+				return true, nil // Deletion marker found, success
+			}
+
+			// Job still exists without deletion marker, continue polling
+			logger.V(1).Info("Job still exists without deletion marker, retrying...",
+				"jobName", jobName, "namespace", jobNamespace)
+			return false, nil
+		})
+
+	if err != nil {
+		if errors.Is(err, context.DeadlineExceeded) {
+			logger.Info("Timeout waiting for job deletion marker, proceeding anyway",
+				"jobName", jobName, "namespace", jobNamespace, "timeout", pollTimeout)
+			return nil // Don't fail the reconciliation on timeout
+		}
+		return fmt.Errorf("error waiting for job deletion marker: %w", err)
+	}
+
+	return nil
+}
+
+// waitForEtcdStatusConditionUpdate waits for the etcd status condition to reflect the expected state.
+// This ensures that the next reconciliation will not see a stale etcd status due to informer cache staleness.
+func (r *Reconciler) waitForEtcdStatusConditionUpdate(ctx context.Context, logger logr.Logger, etcdNamespace, etcdName string, expectedCondition druidv1alpha1.Condition) error {
+	logger.Info("Waiting for etcd status condition to be updated",
+		"etcdName", etcdName, "namespace", etcdNamespace,
```

Comment:
The `logger` object should ideally be created with `etcdName` and `namespace` key-values at the very beginning itself, so we can avoid re-specifying these throughout the reconciler code, similar to https://github.com/gardener/etcd-druid/blob/34d685263a3d89e6cc0802dddc836356b6b1d8fe/internal/controller/etcd/reconcile_spec.go#L29